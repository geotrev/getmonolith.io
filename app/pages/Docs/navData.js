import {
  introductionPath,
  downloadPath,
  brandingPath,
  javascriptPath,
  gridPath,
  typographyPath,
  buttonsPath,
  formsPath,
  modalsPath,
  accordionsPath,
  dropdownsPath,
  tooltipsPath,
  alignmentPath,
  offsetOrderPath,
  textPath,
  displayPath,
  colorPath,
  spacingPath,
} from "routes"

export const NAV_DATA = [
  {
    header: "Overview",
    links: [
      { name: "Introduction", url: introductionPath },
      { name: "Download", url: downloadPath },
      { name: "Branding", url: brandingPath },
      { name: "JavaScript", url: javascriptPath },
    ],
  },
  {
    header: "Layout",
    links: [
      { name: "Flex Grid", url: gridPath },
      { name: "Alignment", url: alignmentPath },
      { name: "Offset / Order", url: offsetOrderPath },
      { name: "Spacing", url: spacingPath },
    ],
  },
  {
    header: "Elements",
    links: [
      { name: "Buttons", url: buttonsPath },
      { name: "Forms", url: formsPath },
      { name: "Typography", url: typographyPath },
    ],
  },
  {
    header: "Components",
    links: [
      { name: "Accordions", url: accordionsPath },
      { name: "Dropdowns", url: dropdownsPath },
      { name: "Modals", url: modalsPath },
      { name: "Tooltips", url: tooltipsPath },
    ],
  },
  {
    header: "Utilities",
    links: [
      { name: "Color", url: colorPath },
      { name: "Display", url: displayPath },
      { name: "Text", url: textPath },
    ],
  },
]
