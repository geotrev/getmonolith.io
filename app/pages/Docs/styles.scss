@import "app/global";

#docs {
  .small.button {
    margin-bottom: 48px;
  }

  .article-wrapper {
    opacity: 0;
    transition: $global-ease 0.35s opacity;

    &.fadeIn {
      opacity: 1;
    }

    pre {
      margin-bottom: 32px !important;
    }

    .filler {
      background: $gray900;
      border-radius: $global-border-radius;
      margin-bottom: 1rem;
    }

    .static-height {
      height: 200px;
    }

    h2,
    h3 {
      margin-top: 54px;
    }

    h4,
    h5,
    h6 {
      margin-top: 32px;
    }

    // override accordion examples
    .accordion-row {
      > h1,
      > h2,
      > h3,
      > h4,
      > h5,
      > h6 {
        margin-top: 0;
      }
    }

    .dropdown-menu {
      h1,
      h2,
      h3,
      h4,
      h5,
      h6 {
        margin-top: 0;
      }
    }

    .inverted-bg {
      background: $gray200;
      padding: 1rem 1rem 0 1rem;
    }

    .filler-bg {
      background: $gray900;
      padding: 1rem;
      margin-bottom: 1rem;
      border-radius: $global-border-radius;

      .filler {
        background: $gray800;
        color: white;
        margin-bottom: 0;
      }
    }
  }
}
