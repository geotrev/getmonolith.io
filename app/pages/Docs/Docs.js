import React from "react"

import SideNav from "app/components/SideNav"
import DocsRoutes from "app/components/DocsRoutes"
import "./styles.scss"

export default function Docs() {
  return (
    <div id="docs" className="fluid grid has-no-padding">
      <div className="row">
        <SideNav />
        <div className="is-xsmall-12 is-xlarge-10 column has-padding-3">
          <DocsRoutes />
        </div>
      </div>
    </div>
  )
}
