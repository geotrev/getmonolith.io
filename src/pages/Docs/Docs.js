import React from "react"
import "./Docs.scss"

import { ScrollUpOnMount } from "helpers"
import { SideNav } from "components"
import DocsRoutes from "./DocsRoutes/DocsRoutes"

const Docs = () => {
  const navItems = [
    {
      header: "Overview",
      links: [
        { name: "Introduction", url: "/docs/overview/introduction" },
        { name: "Download", url: "/docs/overview/download" },
        { name: "Branding", url: "/docs/overview/branding" },
        { name: "JavaScript", url: "/docs/overview/javascript" },
        { name: "Accessibility", url: "/docs/overview/accessibility" },
      ],
    },
    {
      header: "Elements",
      links: [
        { name: "Grid", url: "/docs/elements/grid" },
        { name: "Typography", url: "/docs/elements/typography" },
        { name: "Buttons", url: "/docs/elements/buttons" },
        { name: "Forms", url: "/docs/elements/forms" },
      ],
    },
    {
      header: "Components",
      links: [
        { name: "Modals", url: "/docs/components/modals" },
        { name: "Accordions", url: "/docs/components/accordions" },
      ],
    },
    {
      header: "Utilities",
      links: [
        { name: "Alignment", url: "/docs/utilities/alignment" },
        { name: "Offset / Order", url: "/docs/utilities/offset-order" },
        { name: "Text", url: "/docs/utilities/text" },
        { name: "Display", url: "/docs/utilities/display" },
        { name: "Color", url: "/docs/utilities/color" },
        { name: "Spacing", url: "/docs/utilities/spacing" },
      ],
    },
  ]

  return (
    <div id="docs" className="medium-section fluid grid">
      <ScrollUpOnMount />
      <div className="row">
        <SideNav navListClasses="xsmall-12 columns has-no-padding" navItems={navItems} />

        <div className="xsmall-12 xlarge-10 has-no-padding columns">
          <DocsRoutes />
        </div>
      </div>
    </div>
  )
}

export default Docs
