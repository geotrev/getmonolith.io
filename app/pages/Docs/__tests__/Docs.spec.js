import React from "react"
import Docs from "pages/Docs/Docs"
import { BrowserRouter as Router } from "react-router-dom"

jest.mock("components/DocsRoutes", () => global.simpleMock("DocsRoutes"))
jest.mock("components/SideNav", () => global.simpleMock("SideNav"))

describe("<Docs />", () => {
  it("renders", () => {
    const wrapper = mount(
      <Router>
        <Docs />
      </Router>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
