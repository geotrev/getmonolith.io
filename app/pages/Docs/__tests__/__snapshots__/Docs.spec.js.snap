// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Docs /> renders 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Docs>
      <div
        className="fluid grid has-no-padding"
        id="docs"
      >
        <div
          className="row"
        >
          <SideNav />
          <div
            className="is-xsmall-12 is-xlarge-10 column has-padding-3"
          >
            <DocsRoutes />
          </div>
        </div>
      </div>
    </Docs>
  </Router>
</BrowserRouter>
`;
